armory/datasets/package.py

```diff
@@ -100,21 +128,43 @@ def extract(name, data_dir: str = None, overwrite: bool = False):
     completed_process = subprocess.run(cmd)
     completed_process.check_returncode()
 
-    # should have directory structure <tmp_dir>/<name>/<version>/<data>
-    if len(os.listdir(tmp_dir)) != 1:
+    tmp_dir_listdir = os.listdir(tmp_dir)
+    # should have directory structure <tmp_dir>/<name>/*
+    if len(tmp_dir_listdir) != 1:
         raise ValueError(f"{tmp_dir} has more than 1 directory inside")
-    if name not in os.listdir(tmp_dir):
+    if name not in tmp_dir_listdir:
         raise ValueError(f"{name} does not match directory in {tmp_dir}")
     tmp_dir_name = tmp_dir / name
 
-    if len(os.listdir(tmp_dir_name)) != 1:
-        raise ValueError(f"{tmp_dir_name} has more than 1 directory inside")
-    if version not in os.listdir(tmp_dir_name):
-        raise ValueError(f"{version} does not match directory in {tmp_dir_name}")
-    source_data_dir = tmp_dir_name / version
-
-    if any(child.is_dir() for child in source_data_dir.iterdir()):
-        raise ValueError("Data directory should not have subdirectories")
+    tmp_dir_name_listdir = os.listdir(tmp_dir_name)
+    subdir_parsed_list = [subdir.split("/")[1] for subdir in subdir_list]
+    if set(tmp_dir_name_listdir) != set(subdir_parsed_list):
+        raise ValueError(
+            f"Directories in {tmp_dir_name_listdir} do not match {subdir_parsed_list}"
+        )
+    if len(tmp_dir_name_listdir) > 1:
```

Comment:
the code in `extract()` is becoming a bit difficult to grok. A couple suggestions could be to rename `subdir_list` to `expected_subdir_list` and to add a concise comment before `if len(tmp_dir_name_listdir) > 1` explaining when we expect this to be true, but generally suggest making this a little easier to parse

Reply:
check 1fd925b7